tasks/update_gsheet_task.py

```diff
@@ -0,0 +1,112 @@
+import json
+from typing import List
+
+import luigi as luigi
+from datetime import datetime
+import pandas as pd
+import logging
+
+from backend.data import ExtractCovid19IndiaData
+from backend.repository import GSheetRepository
+from backend.metrics.calculations import impute_hospitalization_percentages, extend_and_impute_metrics
+from tasks import FetchCovid19IndiaDataTask
+
+log = logging.getLogger(__name__)
+
+
+class UpdateGSheetTask(luigi.ExternalTask):
+    worksheet_hospitalizations = 'Phase 2 - Hospitalization'
+    worksheet_districts = 'Phase 2 - Districts'
+    worksheet_states = 'Phase 2 - States'
+
+    states_is_valid = False
+    districts_is_valid = False
+
+    metrics_needed_by_dashboard = [
+        # updated 2020-12-28 Phase 1
+        # more can be found here https://github.com/swb-ief/etl-pipeline/blob/data_pipeline_readme/explainers/WorkflowDescription.md
+        # note it is still on a branch. soon replace /data_pipeline_readme/ for /master/
+
+        'delta.confirmed',
+        'delta.deceased',
+        'delta.tested',
+        'delta.recovered',
+        'delta.hospitalized',
+        'delta.percent.case.growth',
+        'delta.positivity',
+
+        'spline.active',
+        'spline.deceased',
+        'spline.hospitalized',
+        'spline.recovered',
+
+        'total.deceased.levitt',  # this needs a better name
+
+        'MA.21.daily.tests',
+        'MA.21.delta.positivity',
+    ]
+    # Not yet added R generated metrics mean.mean, CI_lower.mean, CI_upper.mean, doubling.time
+
+    state_keys = ['date', 'state']
+    district_keys = ['date', 'state', 'district']
+
+    state_columns_needed_by_dashboard = state_keys + metrics_needed_by_dashboard
+    district_columns_needed_by_dashboard = district_keys + metrics_needed_by_dashboard
+
+    def run(self):
+        # we are skipping older data since we only have low case numbers there.
+        start_date = datetime(2020, 4, 1)
+
+        repository = GSheetRepository(GSheetRepository.get_worksheet_url_from_env())
+
+        fetch_covid19_india_task = yield FetchCovid19IndiaDataTask()
+
+        # Kick off Ward data collection trough tasks
+        # See if luigi can parallelize these 'download' steps
```

Comment:
is this where the ward data collection task will go?

Reply:
yes, it would be 1 task, but it might have a luigi sub task for each of the different cities depending on our discussions on slack.
and luigi should be able to parallelize the covid19inda.org api download and the ward downloads and do retries where needed in parallel, once they are done we do the imputation's and calculations we need for the dashboard on that data.